core-processor/src/main/java/io/micronaut/inject/ast/utils/EnclosedElementsQuery.java

```diff
@@ -53,7 +53,8 @@
 @Internal
 public abstract class EnclosedElementsQuery<C, N> {
 
-    private final Map<N, Element> elementsCache = new ConcurrentLinkedHashMap.Builder<N, Element>().maximumWeightedCapacity(200).build();
+    private static final int MAX_ITEMS_IN_CACHE = 200;
+    private final Map<N, Element> elementsCache = new LinkedHashMap<>();
```

Comment:
I had to replace `ConcurrentLinkedHashMap` as it always allocates some internal data structures multiplied by the processors count. That would crash on the machine I have to run tests.

Reply:
fyi you could change that to a reasonable number since it is an internal class, e.g. 4. It was all redone in Caffeine to be dynamically sized based on contention but this older utility was copied in since the team wanted something embedded.